lib/chef/resource/macos_userdefaults.rb

```diff
@@ -89,10 +90,11 @@ class MacosUserDefaults < Chef::Resource
 
       property :type, String,
         description: "The value type of the preference key.",
+        deprecated: true,
         equal_to: %w{bool string int float array dict},
         desired_state: false
 
-      property :user, String,
+      property :user, [String, Symbol],
```

Comment:
even if the lib takes it a symbol it doesn't make sense for the user to pass it this way